### PR TITLE
Added the Drainable Class

### PR DESCRIPTION
**Pull request by JacobLD**
This class extends a sprite and requires the sprite to use the DrainableMaterial.

This class exposes a few variables to play with including min_drain_time, max_drain_time, zero_distance_drain, and max_distance_drain.

This class also exposes one function: on_drain_tick. And one signal: Drained; which emits the amount of color that was drained.

Changes that carry impact are all files outside of my testing folder which includes changes to the scripts and materials folder.

### CODE REVIEW

File: scenes/JacobTestScenes/DrainTester.gd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+extends Node2D
+
+
+onready var drainable : Drainable = $FullColor
+var drain_radius = 100
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
```
Comment (AptosGames) [inline scenes/JacobTestScenes/DrainTester.gd:8]: rest of the PR looks great. only concern here is variable framerate! we can check it out on itch and see if it runs the same. i have a 144hz monitor and ive noticed issues using `_process` before

**Review by AptosGames — APPROVED** [review]

**Review by AptosGames — APPROVED** [review]

**Comment by AptosGames** [comment]
not sure why my review isnt unblocking u!

**Comment by JacobLD** [comment]
I added you to a team and gave 'maintain' permissions so that may have done the trick.

File: scenes/JacobTestScenes/DrainTester.gd (hunk 1 above)
Comment (JacobLD) [reply scenes/JacobTestScenes/DrainTester.gd:8]: What types of issues are you talking about? I've never noticed issues on my past projects when it came to my 144hz monitor. Are you referring to issues surrounding the difference between the physics update rate nad the process loop?

**Pull request merged**